roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -154,8 +233,194 @@ def get_num_channels(self):
+        """
+        if 'Ophys' not in metadata:
+            metadata['Ophys'] = {}
+
+        if 'Ophys' not in metadata or 'TwoPthotonSeries' not in metadata['Ophys']:
+            metadata['Ophys']['TwoPhotonSeries'] = [{'name': 'TwoPhotonSeries',
+                                                     'description': 'optical_series_description'}]
+        # Tests if ElectricalSeries already exists in acquisition
+        nwb_es_names = [ac for ac in nwbfile.acquisition]
+        opts = metadata['Ophys']['TwoPhotonSeries'][0]
+        if opts['name'] not in nwb_es_names:
+            # retrieve device
+            device = nwbfile.devices[list(nwbfile.devices.keys())[0]]
+
+            # create optical channel
+            if 'OpticalChannel' not in metadata['Ophys']:
+                metadata['Ophys']['OpticalChannel'] = [{'name': 'OpticalChannel',
+                                                        'description': 'no description',
+                                                        'emission_lambda': 500.}]
+
+            optical_channel = OpticalChannel(**metadata['Ophys']['OpticalChannel'][0])
+            # sampling rate
+            rate = float(imaging.get_sampling_frequency())
+
+            if 'ImagingPlane' not in metadata['Ophys']:
+                metadata['Ophys']['ImagingPlane'] = [{'name': 'ImagingPlane',
+                                                      'description': 'no description',
+                                                      'excitation_lambda': 600.,
+                                                      'indicator': 'Indicator',
+                                                      'location': 'Location',
+                                                      'grid_spacing': [.01, .01],
+                                                      'grid_spacing_unit': 'meters'}]
+            imaging_meta = {'optical_channel': optical_channel,
+                            'imaging_rate': rate,
+                            'device': device}
+            metadata['Ophys']['ImagingPlane'][0] = update_dict(metadata['Ophys']['ImagingPlane'][0], imaging_meta)
+
+            imaging_plane = nwbfile.create_imaging_plane(**metadata['Ophys']['ImagingPlane'][0])
+
+            # def data_generator(imaging, channels_ids):
+            #     #  generates data chunks for iterator
+            #     for id in channels_ids:
+            #         data = recording.get_traces(channel_ids=[id]).flatten()
+            #         yield data
+            #
+            # data = data_generator(imaging=imaging, channels_ids=curr_ids)
+            # ophys_data = DataChunkIterator(data=data, iter_axis=1)
+            acquisition_name = opts['name']
+
+            # using internal data. this data will be stored inside the NWB file
+            ophys_ts = TwoPhotonSeries(
+                name=acquisition_name,
+                data=imaging.get_video(),
```

Comment:
We should compress and use iterative write for data here

Reply:
in `NwbRecordingExtractor` we create a data chunk iterator that iterates over the channel_ids. Should we split this in time chunks instead?

Reply:
https://pynwb.readthedocs.io/en/stable/tutorials/general/iterative_write.html